### DISCUSSION TITLE
What would it take to create our own font?

### DISCUSSION THREAD

**Discussion by lipis** [excalidraw/excalidraw#2987] [General]
Read this comments for more https://github.com/excalidraw/excalidraw/issues/2945#issuecomment-774680218

[Calligraphr](https://www.calligraphr.com/en/) seems like a tool we can create our own font. What do we need in order to achieve the following:

1. Public repo (we can host it under `excalidraw` org)
2. Source files of the latin, numbers and basic chars
3. GitHub actions, generating the needed outputs for consuming it in the web
4. How easily can we add more languages with the same _style_?
5. Can we start by importing Virgil font? or any other font we choose that have the correct license?

By doing that we are in full control at any time to fix and support whatever seems broken.

**Comment by dwelle** [comment]
Haven't tried out the Calligraphr so maybe it makes it super easy, but I once tried modifying a single character SVG in a vector editor and I quickly gave up — it was mainly because (as this [blog post points out](https://www.bapt.xyz/b/handwriting)) most fonts don't use paths for glyphs, but shapes, so editing them (or creating them) isn't easy.

![image](https://user-images.githubusercontent.com/5153846/107151366-842a2c80-6962-11eb-8fd8-09c68eb697db.png)

So if it comes to creating SVG shapes, I reckon what it takes is someone who already has experience making fonts.

**Reply by lipis** [reply]
Yeah.. I've imagined that, it would take someone that have done it before in the past. Font is no easy subject for sure.. Maybe we should find one that already open source and exists to begin with.. but haven't seen anything in the handwritten format.

**Comment by xixixao** [comment]
I disagree with the original comment https://github.com/excalidraw/excalidraw/issues/2945#issuecomment-774680218 that character randomization is **quintessential**. It might have some pros (looking more like whiteboard) and some cons (perf, legibility). 

(this is perhaps orthogonal to this discussion, but I wanted to point this out without spamming the original issue)

**Reply by eddyg** [reply]
Perhaps _quintessential_ was the wrong word. My point was that when combined with the "Artist" and "Cartoonist" Sloppiness factors (and to a lesser extent, "Architect") having a font with letter variability would be preferable to one where every occurrence of a letter looking exactly the same. There is no reason having variants of an "e" or an "s" should affect legibility.

FWIW, [this article](https://jvns.ca/blog/2020/08/08/handwritten-font/) has more information on how this is accomplished.

Also, [these fonts from RedHat](https://github.com/RedHatOfficial/bwht-fonts) **may** be worth looking at for inspiration? _(emphasis mine)_:
> We sourced handwriting from friends and colleagues, working to capture the themes of the piece: feminism, hard work, boldness, and transparency among others. From those forms, we shaped and combined, extrapolated more, and built full character sets. **And then, to ensure a natural and organic feeling in each line, we added Open Type features to help "randomize" the flow.**
